server/base/src/main/java/org/apache/accumulo/server/init/Initialize.java

```diff
@@ -362,7 +361,7 @@ private boolean initialize(SiteConfiguration siteConfig, Configuration hadoopCon
     UUID uuid = UUID.randomUUID();
     // the actual disk locations of the root table and tablets
     String[] configuredVolumes = VolumeConfiguration.getVolumeUris(siteConfig, hadoopConf);
-    VolumeChooserEnvironment chooserEnv = new VolumeChooserEnvironmentImpl(ChooserScope.INIT, null);
+    VolumeChooserEnvironment chooserEnv = new InitVolumeChooserEnvironmentImpl(siteConfig);
```

Comment:
If passing a keyextent could do the following here.

```suggestion
    VolumeChooserEnvironment chooserEnv = new InitVolumeChooserEnvironmentImpl(siteConfig, RootTable.EXTENT);
```

Reply:
Thanks for the review, @keith-turner! If we end up implementing an InitServerContext as per Christopher's suggestion, then I think we won't need to introduce an InitVolumeChooserEnvironmentImpl. In that case, do you think we should improve the VolumeChooserEnvironmentImpl to have a table ID and endRow for the INIT scope?

Reply:
I am not sure, because I have not looked into the details enough. Would the user have to ensure they provide a chooser to init that does not call these methods?

Reply:
I looked into this some more and I see that those three methods are currently related to the table scope.  Seems like the init scope may only be used for tables.  If so maybe its ok to make the info available w/ an update to the javadocs to clarify the init scope is more like init_table.  Could also update the javadocs for the three methods in VolumeChooserEnvironment that return tablet info to say they can only be called for TABLE or INIT scope.  This could be follow on work, it seems like the information could be useful to some chooser impls.

Reply:
The INIT scope is for all initial tables during the INIT phase... since those tables can't be individually configured in ZK yet and our config isn't expressive enough to distinguish between the different tables whose initial files are created during INIT. So, I never thought a chooser would be able to (or care to) make different choices for different tables during INIT. However, I wasn't thinking about the possibility of having a chooser make hard-coded decisions, or use some external configuration system (Java system props or ENV variables). So, I agree that adding table information to the VolumeChooserEnvironment may be useful as follow-on work to support some chooser implementations.